app/models/report.rb

```diff
@@ -32,6 +32,14 @@ def object_type
     :flag
   end
 
+  def from_domain
+    Report.where(account: Account.by_domain(account.domain))
+  end
+
+  def targeting_domain
+    Report.where(target_account: Account.by_domain(account.domain))
+  end
```

Comment:
I am worried about the performances here, but I'm no Rails expert, so I'll let someone more knowledgeable comment.

Reply:
Yeah, it'd probably be better to include `domain` on the Report object, such that we could just do `Report.where(target_domain: domain).count` and `Report.where(domain: domain).count`

But this would require database changes, and be backwards incompatible.

Reply:
This becomes a `SELECT ... WHERE id IN (SELECT ... )` query which is fine, it's not like all accounts are loaded into Rails before being passed into the where clause